### PR TITLE
new module: srst2/srst2

### PR DESCRIPTION
**Pull request by jvhagey**
Closes #[1566](https://github.com/nf-core/modules/issues/1566)

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/srst2/srst2/meta.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+keywords:
+  - mlst
+  - typing
+  - illumina
+tools:
+  - srst2:
+      description: "Short Read Sequence Typing for Bacterial Pathogens"
+      homepage: { http://katholt.github.io/srst2/ }
+      documentation: { https://github.com/katholt/srst2/blob/master/README.md }
+      tool_dev_url: { https://github.com/katholt/srst2 }
```
Comment (sateeshperi) [inline modules/srst2/srst2/meta.yml:15]:
```suggestion
      homepage: "http://katholt.github.io/srst2/"
      documentation: "https://github.com/katholt/srst2/blob/master/README.md"
      tool_dev_url: "https://github.com/katholt/srst2"
```

File: modules/srst2/srst2/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    script:
+    def args = task.ext.args ?: ""
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def read_s = meta.single_end ? "--input_se ${fastq_s}" : "--input_pe ${fastq_s[0]} ${fastq_s[1]}"
+    if (meta.db=="gene") {
+        database = "--gene_db ${db}"
+    } else if (meta.db=="mlst") {
+        database = "--mlst_db ${db}"
+    } else {
+        return
```
Comment (sateeshperi) [inline modules/srst2/srst2/main.nf:32]: don't think you need to return here in groovy

Comment (jvhagey) [reply modules/srst2/srst2/main.nf:32]: I was trying to get it to exit with an error. If you don't pick one of those two options nothing will happen so I was trying to get it to stop. Is there a better way to do it?

Comment (sateeshperi) [reply modules/srst2/srst2/main.nf:32]: do you have tests for the `mlst` db?

Comment (Emiller88) [inline modules/srst2/srst2/main.nf:32]:
```suggestion
        error "Please set meta.db to either \"gene\" or \"mlst\""
```

File: modules/srst2/srst2/meta.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - txt:
+      type: file
+      description: results text file
+      pattern: "*_fullgenes_*_results.txt"
+  - txt:
+      type: file
+      description: results text file
```
Comment (Emiller88) [inline modules/srst2/srst2/meta.yml]: Same as above

File: tests/modules/srst2/srst2/main.nf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+workflow test_srst2_srst2_single_end {
+
+    input = [ 
+        [ id:'test', single_end:true, db:"gene" ], // meta map
+        file(params.test_data['bacteroides_fragilis']['illumina']['test1_1_fastq_gz'], checkIfExists: true),
+        file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/srst2/resFinder_20180221_srst2.fasta')  // Change to params.test_data syntax after the data is included in tests/config/test_data.config
+    ]
+
+    SRST2_SRST2(input)
+}
```
Comment (Emiller88) [inline tests/modules/srst2/srst2/main.nf:28]: Let's add some tests is possible for the mlst db from @sateeshperi's review

**Review by Emiller88 — CHANGES_REQUESTED** [review]
- [x] Clear up the difference between the results txt files
- [x] Add tests fir mlst db or provide an explaination of why that's not possible
- [x] If it's not possible, add a test to check that the logic functions
- [ ] Add tests for passing something other than `gene` or `mlst`(See [pytest-workflow examles](https://pytest-workflow.readthedocs.io/en/stable/#test-options) for writing a failing test)

File: tests/modules/srst2/srst2/test.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+- name: srst2 srst2 test_srst2_srst2_exit #Testing pipeline exit when not meta.db
```
Comment (Emiller88) [inline tests/modules/srst2/srst2/test.yml:1]:
```suggestion
- name: srst2 srst2 test_srst2_srst2_exit #Testing pipeline exit when not meta.db
  command: nextflow run tests/modules/srst2/srst2 -entry test_srst2_srst2_exit -c tests/config/nextflow.config
```
The `command` is missing so pytest workflow doesn't have anything to run.

Comment (jvhagey) [reply tests/modules/srst2/srst2/test.yml:1]: @Emiller88 Looks like it I got it to work, thanks for the fix! Let me know if anything else is required.

**Review by Emiller88 — APPROVED** [review]
Awesome work!

**Review by sateeshperi — APPROVED** [review]
gr8 job! and congrats on your first nf-core module 🥇

**Pull request merged**